mitodl/micromasters
# Moved dashboard API call to DashboardPage

#### What are the relevant tickets?
Fixes #1655

#### What's this PR do?
Delays loading the dashboard API until the user looks at the dashboard page. The user will load the dashboard API when they click next and enroll in a program, but this shouldn't block anything on their end.

#### How should this be manually tested?
Load `/profile` and verify that no request to `/api/v0/dashboard/` is loaded and that the user still sees programs they can enroll in.

Also, go to `/settings` and refresh the browser. Verify that `/api/v0/dashboard/` is not loaded. Then click the logo on the upper left to direct you to the dashboard. Verify that the dashboard loaded properly.
